tests/test_all_rules.py

```diff
@@ -438,30 +438,44 @@ def test_deprecated_rules(self):
     def test_integration(self):
         """Test that rules in integrations folders have matching integration defined."""
         failures = []
+        index_pattern_integrations = ["apm", "endpoint", "windows", "winlog"]
```

Comment:
Rules with integration tags from this list are rules where we have to do unit testing based on index patterns specified in the rule, not the query itself.

Reply:
Nit: Should this list match the same list as here? https://github.com/elastic/detection-rules/blob/e70b0ab44946014067b166d2013bf8aba51471ea/detection_rules/rule.py#L957

We may want to pull this list out to a constant.

e.g. `NON_DATASET_PACKAGES = ("apm", "endpoint", "windows", "winlog")`

Reply:
Changed the name of the variable. Unfortunately I dislike storing these as static values in the unit test but I am not aware of how to dynamically identify them. 

The lists will not be the same. One is referencing integration tags, and the other is a substring of index patterns. `logs-windows-*` and `logs-winlogbeat-*` are two common index patterns for Windows logs with the Windows integration.